extensions/parquet/compression/src/main/java/io/deephaven/parquet/compress/CompressorAdapter.java

```diff
@@ -46,6 +47,63 @@ public void reset() {}
         public void close() {}
     };
 
+    /**
+     * Reads exactly {@code length} number of bytes from the input stream into the provided byte array.
+     *
+     * @param in The input stream to read from
+     * @param bytes The byte array to read into
+     * @param offset The offset in the byte array to start reading at
+     * @param length The number of bytes to read
+     * @throws IOException If an error occurs while reading from the input stream
+     * @throws IllegalArgumentException If the byte array is too small to read {@code length} number of bytes
+     * @throws IOException If the expected number of bytes could not be read
+     */
+    static void readNBytes(final InputStream in, final byte[] bytes, final int offset, final int length)
+            throws IOException {
+        if (length > bytes.length - offset) {
+            throw new IllegalArgumentException("Bytes array of length " + bytes.length + " with offset " + offset +
+                    " is too small to read " + length + " bytes");
+        }
+        readNBytesHelper(in, bytes, offset, length);
+    }
+
+    /**
+     * Reads exactly {@code length} number of bytes from the input stream into a new byte array and returns it.
+     *
+     * @see #readNBytesHelper(InputStream, byte[], int, int)
+     */
+    static byte[] readNBytes(final InputStream in, final int length) throws IOException {
+        final byte[] bytes = new byte[length];
+        readNBytesHelper(in, bytes, 0, length);
+        return bytes;
+    }
+
+    private static void readNBytesHelper(final InputStream in, final byte[] bytes, final int offset, final int length)
+            throws IOException {
+        int numRead = 0;
+        while (numRead < length) {
+            final int count = in.read(bytes, offset + numRead, length - numRead);
+            if (count == -1) {
+                throw new IOException("Expected to read " + length + " bytes, but read " + numRead + " bytes");
+            }
+            numRead += count;
+        }
+    }
+
+    /**
+     * An {@link InputStream} that will not close the underlying stream when {@link InputStream#close()} is called.
+     */
+    final class InputStreamNoClose extends FilterInputStream {
```

Comment:
Can be static, yeah?

Reply:
I tried that but IntelliJ suggested "Modifier 'static' is redundant for inner classes of interfaces". So removed it.